### ISSUE TITLE
[v2] No easy way to opt out of tree sitter parsers

### ISSUE THREAD

**Issue by mgmarlow** [SystemCrafters/crafted-emacs#366]
Hello! I'm on the RC1 branch on Windows 11 and Emacs 29.1. After configuring `crafted-ide` I ran into an issue where the `protobuf` parser failed to install (for some unknown reason). Since `crafted-ide-config` uses `treesit-auto-install-all`, every time I closed and reopened Emacs, I'd once again be prompted to install the missing tree sitter parser (protobuf) which would fail to install in the same way.

To work around this, I manually require `treesit-auto` and set the `treesit-auto-opt-out-list` before continuing with `crafted-ide-config`.

Perhaps Crafted Emacs should support its own method to avoid this install-loop and opt out of certain parsers?

```elisp
;; ...
(require 'crafted-ide-packages)

(package-install-selected-packages :noconfirm)

;; Require treesit-auto before loading ide-config so I can opt out of protobuf:
(require 'treesit-auto)
(add-to-list 'treesit-auto-opt-out-list 'protobuf)
(require 'crafted-ide-config)
```

**Comment by jvdydev** [comment]
Thank you for trying out the RC1.

I understand that this can be quite annoying since it re-prompts on every boot if at least one grammar is not installed (it seems protobuf doesn't compile for me either :smiling_face_with_tear: ).

I think the best solution would be to make tree-sitter opt-in by default (e.g. with a `crafted-ide-configure-treesitter` function) which could take such a "opt-out" list of parsers.

Something like:
```emacs-lisp
(require 'crafted-ide-packages)

(package-install-selected-packages :noconfirm)

(require 'crafted-ide-config)
(crafted-ide-configure-treesitter '(protobuf)) ; configures treesitter, but without protobuf
```

This would also allow separating out the way to set up tree-sitter for Emacs pre-29 vs. 29-onwards.
Additionally it works around another limitation since tree-sitter requires a C compiler (which automatically makes Crafted Emacs have it as a requirement as well).

**Comment by mgmarlow** [comment]
Agreed with the sentiment around requiring a C compiler, particularly for windows. I had previously set up tree sitter for myself before trying RC1 so I had WinGW ready to go. 

I would personally prefer an opt-in list as I really only care about tree sitter for a few languages, and it reads better from a configuration standpoint (the way the opt-out list is specified appears that protobuf is the only parser being installed, though the opposite is true). That's a minor nitpick though. 

Given the extra requirements around tree sitter, I wonder whether moving it to a separate module could make sense? That way the documentation could be clear that tree sitter = C compiler needed + installation of extra parsers.

**Comment by jvdydev** [comment]
I think an extra module is a bit overkill as the configuration itself is only a few lines of code (see changes in #367).
Extra documentation for requiring a C compiler to install the language grammar would make sense.

Regarding the opt-out list, I thought the same at first, but figured that since it can be called without any arguments, it may be a bit more obvious it's opt-out (it is also documented as such in-line and in the docs, see #367).

**Comment by jeffbowman** [comment]
@mgmarlow I have merged the fix @jvdydev provided in #367, if you could pull the updates and test again that would be appreciated.  Please review the docs (and code if you are comfortable with that) and let us know.  Thanks for reporting this issue!

**Comment by mgmarlow** [comment]
Works great, thanks!
